### PR TITLE
Feature : Yolo v8 Seg Integration

### PR DESCRIPTION
**Pull request by kshitijrajsharma**
## What does this PR do ? 

- Introduces YOLOv8 Segmentation  integration 
https://docs.ultralytics.com/models/yolov8/ 

What are v1 and v2 of yolo v8 segmentation ? 
- v1 is yolo trained from scratch with RAMP data , Supports multimasks 
- v2 is yolo finetuned from pretrained model with RAMP data , Only supports binary masks ( Omdeena Research output integration ) 

- Preprocessing , Training and inference code is available for both of the version 

- Ultralytics and pytorch is added ! 

**Important upgrade :** 
- Upgraded dependencies so that it can work without installing ramp dependencies 

More information on. the research : https://github.com/hotosm/omdena-hot-2024/ 

## How to test ? 
Run `python test_yolo_v1.py` and `python test_yolo_v2.py` , Both of them should pass !  Also test running ramp training too , it should work as well

### CODE REVIEW

**Review by omranlm — APPROVED** [review]
Great and efficient.

let's do more testing using the API and new UI
